src/internal/kopia/conn.go

```diff
@@ -389,14 +386,19 @@ func checkCompressor(compressor compression.Name) error {
 		}
 	}
 
-	return errors.Errorf("unknown compressor type %s", compressor)
+	return clues.Stack(clues.New("unknown compressor type"), clues.New(string(compressor)))
 }
 
 func (w *conn) LoadSnapshots(
 	ctx context.Context,
 	ids []manifest.ID,
 ) ([]*snapshot.Manifest, error) {
-	return snapshot.LoadSnapshots(ctx, w.Repository, ids)
+	mans, err := snapshot.LoadSnapshots(ctx, w.Repository, ids)
+	if err != nil {
+		return nil, clues.Stack(err).WithClues(ctx)
```

Comment:
Just wondering if we should have  the `if err != nil` check withing the `Stack` call? This might simplify the logic in a few places. Just as a reference [pkg/errors:Wrap](https://github.com/pkg/errors/blob/5dd12d0cfe7f152f80558d591504ce685299311e/errors.go#L184-L187) was doing it.

Reply:
Not a bad idea, but we'd have to keep this pattern here anyway.  The trouble isn't `clues.Stack()`, it's `.WithClues(ctx)`.  If we call `clues.Stack(nil).WithClues(ctx)` we'd segfault.